src/memory_map.rs

```diff
@@ -91,26 +97,42 @@ pub enum MemoryAreaType {
     Defective,
 }
 
-/// An iterator over Available memory areas.
+/// An iterator over all memory areas
 #[derive(Clone, Debug)]
-pub struct MemoryAreaIter<'a> {
+pub struct AllMemoryAreaIter<'a> {
```

Comment:
`EFIMemoryMap` also could use an iterator over all memory areas.

Reply:
It looks to me like EFIMemoryMapTag already iterates over all memory areas.